src/endpointDefinitions.ts

```diff
@@ -24,16 +24,17 @@ export enum DefinitionType {
   mutation = 'mutation',
 }
 
-type GetResultDescriptionFn<EntityTypes extends string, ResultType, QueryArg> = (
-  result: ResultType,
+type GetResultDescriptionFn<EntityTypes extends string, ResultType, QueryArg, ErrorType> = (
+  result: ResultType | undefined,
+  error: ErrorType | undefined,
   arg: QueryArg
 ) => ReadonlyArray<EntityDescription<EntityTypes>>;
```

Comment:
Up for discussion: instead of going `(result, arg) => Entities` to `(result?, error?, arg) => Entities`, we could also go for

* `(arg, result?, error?) => Entities`

or

* `({result, error, arg}) => Entities`

Reply:
i think having the original input last makes sense so it follows the whole `acc, curr, array` type of signature?